src/auth-socket-middleware.ts

```diff
@@ -19,17 +20,9 @@ export default function registerAuthSocketMiddleware(io: Server) {
       }
       logger.info(`Token decoded: ${JSON.stringify(decoded, null, 2)}`);
 
-      if (
-        decoded && 
-        typeof decoded !== 'string' && 
-        decoded.payload && 
-        decoded.payload.uuid
-      ) {
-        const uuid = decoded.payload.uuid;
-        logger.info(`user: ${uuid} is listening notifications`);
-        socket.join(uuid);
-      } else {
-        const email = typeof decoded === 'string' ? decoded : decoded!.email;
+      const jwtPayload = (decoded as Jwt)?.payload;
```

Comment:
why this casting is needed?

Reply:
No, it is not necessary. I will leave the condition for using the email as it was.